core/vm/evm.go

```diff
@@ -194,7 +194,12 @@ func (evm *EVM) call(typ OpCode, caller ContractRef, addr libcommon.Address, inp
 	p, isPrecompile := evm.precompile(addr)
 	var code []byte
 	if !isPrecompile {
-		code = evm.intraBlockState.GetCode(addr)
+		code = evm.intraBlockState.ResolveCode(addr)
+		if depth == 0 && typ == CALL {
+			if dd, ok := evm.intraBlockState.GetDelegatedDesignation(addr); ok {
+				evm.intraBlockState.AddAddressToAccessList(dd)
```

Comment:
Why do we need this? Isn't the delegated designation already added to the access list by `gasCallEIP7702`?

Reply:
this scenario is when the `tx.to` has a DD. This is a top level "call" (depth==0) for which dynamic cost isn't calculated through gasCallEIP7702 (which is called only when call is done via evm opcode), and so DD is not added to access_list.
Also, the resolution here doesn't cost anything (the intrinsic cost is assumed to cover this). Only when resolution is while executing some message call evm opcode, does the warm/cold cost get added to dynamic cost.

Reply:
I see. It's better to move this to `IntraBlockState.Prepare`, which handles the addition of the `tx.to` to the access list.